src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+
+    public FigureType getRandomFigureType() {
+        int index = random.nextInt(FigureType.values().length);
+        return FigureType.values()[index];
+    }
+
+    public Figure getRandomFigure(FigureType figureType) {
+        switch (figureType) {
+            case SQUARE:
+                return new Square(random.nextInt(50));
```

Comment:
let's create variable for storing `random.nextInt(50)`, so you could pass it to every constructor